pkg/providers/imagefamily/resolver.go

```diff
@@ -37,8 +37,20 @@ import (
 	"sigs.k8s.io/karpenter/pkg/cloudprovider"
 )
 
-// Resolver is able to fill-in dynamic launch template parameters
-type Resolver struct {
+type Resolver interface {
+	Resolve(
+		ctx context.Context,
+		nodeClass *v1alpha2.AKSNodeClass,
+		nodeClaim *karpv1.NodeClaim,
+		instanceType *cloudprovider.InstanceType,
+		staticParameters *template.StaticParameters) (*template.Parameters, error)
+}
+
+// assert that defaultResolver implements Resolver interface
+var _ Resolver = (*defaultResolver)(nil)
```

Comment:
minor, style, feel free to ignore: I find the cleaner way to write this is:
```
var _ Resolver = &defaultResolver{}
```

Note that this _looks_ like it has an allocation, but it actually doesn't - the compiler is smart enough to realize that the `_` variable here is unused and so it typechecks this expression at compile-time but it doesn't actually allocate anything.

Reply:
I was copying the style I found in a different location of the codebase. It does seem like a weird split where we are using the style you mentioned in the testing packages, but this style elsewhere. I do like the style you mentioned more, and since we are using it in the testing packages, I think I'll switch it.